### DISCUSSION TITLE
Custom PDF generation

### DISCUSSION THREAD

**Discussion by athira-aot** [AOT-Technologies/forms-flow-ai#1619] [Q&A] [answered]
I have an Employee Feedback Form with fields such as name, email, department, position, hiring date, supervisor name, email, phone number, feedback, etc. When I attempt to export a PDF, the generated PDF contains all the fields. Is there a way to customize the PDF generation process so that I can choose to export only selected fields? Additionally, I would like to know how to customize the design to the generated PDF.

**Comment by Raveena-aot** [comment, marked as answer]
Yes, it can be accomplished by generating PDFs using a custom theme.

The formsflow document API serves the purpose of generating PDFs. The document API accepts the 'template' attribute within the request body. The expected value for the 'template' attribute is a Base64-encoded Jinja template. For additional information on the document API, please refer to the following link:
https://github.com/AOT-Technologies/forms-flow-ai/blob/develop/forms-flow-documents/README.md#usage-guidlines

To tailor the PDF generation process according to your needs, please follow these steps:
Begin with a sample form that you wish to export as a PDF.

Here is a sample form that we export as a PDF:
![unnamed](https://github.com/AOT-Technologies/forms-flow-ai/assets/96716528/7b36df23-d0ba-4753-84ad-134bc3ea1055)

- To export specific fields such as Name, EmployeeId, and Department from the form, you'll need to make adjustments to the template as outlined  in the provided link

   https://gist.github.com/Raveena-aot/bfd691adafdc837dc8ccd0977a3fb535

- Encode the template using a base64 encoder of your choice. (  [Base64 Encode and Decode - Online](https://www.base64encode.org/))
- Include the encoded template in the body of the Document API’s  POST request.
  [http://localhost:5006/document/form/{formid}/submission/{submissionId}/export/pdf](http://localhost:5006/document/form/form/%7Bformid%7D/submission/%7BsubmissionId%7D/export/pdf))

- Don't forget to include the Keycloak token in the authorization header for authentication.

Sample request body: 
https://gist.github.com/Raveena-aot/5391c042d053b22c29dc8c773d8d611b


![unnamed](https://github.com/AOT-Technologies/forms-flow-ai/assets/96716528/75d49682-1ba9-4153-8fa8-b6aeedc8453e)

With these steps, you can now effectively customize your PDF generation process. If you have any questions or concerns, please feel free to ask.
